### PR TITLE
chore: Update CLI to utilize new error object.

### PR DESCRIPTION
**Pull request by joe-yeager**
## Description and Context

Depends on https://github.com/HubSpot/hubspot-local-dev-lib/pull/159

- Update the CLI to use the new util methods and error object
- Simplify the logging down to a single method

## Who to Notify

@brandenrodgers @camden11 @kemmerle

### CODE REVIEW

**Review by camden11 — APPROVED** [review]
Looks like this still needs a LDL version bump, but otherwise LGTM

**Review by kemmerle — APPROVED** [review]

File: packages/cli/lib/errorHandlers/apiErrors.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
 const {
   isMissingScopeError,
   isApiUploadValidationError,
-  getAxiosErrorWithContext,
+  getHubSpotHttpErrorWithContext,
```
Comment (brandenrodgers) [inline packages/cli/lib/errorHandlers/apiErrors.js:6]: I know this is constantly in flux, but the current version of this in local-dev-lib is `getUserFriendlyHttpErrorMessage`

File: packages/cli/lib/errorHandlers/apiErrors.js [hunk 2]
```diff
@@ -53,9 +56,9 @@ function logValidationErrors(error, context) {
  * @param {ApiErrorContext}          context
  */
 function logApiErrorInstance(error, context) {
-  if (error.isAxiosError) {
+  if (isHubSpotHttpError(error)) {
     if (overrideErrors(error)) return;
-    const errorWithContext = getAxiosErrorWithContext(error, context);
+    const errorWithContext = getHubSpotHttpErrorWithContext(error, context);
     logger.error(errorWithContext.message);
```
Comment (brandenrodgers) [inline packages/cli/lib/errorHandlers/apiErrors.js:62]: Just noting b/c I ran into this during my testing, `getUserFriendlyHttpErrorMessage` returns the error message so there's no need to use `.message` here

**Comment by joe-yeager** [comment]
I bumped this back to a draft.  I'm going to update this to use all of the changes from the final LDL refactor PR over [here](https://github.com/HubSpot/hubspot-local-dev-lib/pull/159)

File: packages/cli/lib/errorHandlers/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  isSystemError,
+  isFileSystemError,
+} = require('@hubspot/local-dev-lib/errors/index');
+const { shouldSuppressError } = require('./suppressError');
+const { i18n } = require('../lang');
+const util = require('node:util');
+
+const i18nKey = 'lib.errorHandlers.index';
+
+function logError(error, context = {}) {
```
Comment (joe-yeager) [inline packages/cli/lib/errorHandlers/index.js:13]: This is now the singular `logError` method that will handle the logging.  So we can call this and let it determine how to log the output.

File: packages/cli/lib/errorHandlers/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  debugError(error, context);
+
+  if (shouldSuppressError(error)) {
+    return;
+  }
+
+  if (isHubSpotHttpError(error) || isFileSystemError(error)) {
+    logger.error(error.toString());
```
Comment (joe-yeager) [inline packages/cli/lib/errorHandlers/index.js:21]: The `HubSpotHttpError` class has a `toString` method that [returns a friendly string representation of the contents](https://github.com/HubSpot/hubspot-local-dev-lib/pull/159/files#diff-ae25cee36b980aa393de912585da6d585d226617b9595f4a65e6e3215c2a1925R65).

I am add a similar method for the `FileSystemError` class so this method can be naive about how to format the messages for that class.

**Pull request merged**